src/workerd/jsg/modules.h

```diff
@@ -195,53 +219,116 @@ class ModuleRegistryImpl final: public ModuleRegistry {
   }
 
   void add(kj::Path& specifier, ModuleInfo&& info) {
-    entries.insert(Entry(specifier, kj::fwd<ModuleInfo>(info)));
+    entries.insert(Entry(specifier, Type::BUNDLE, kj::fwd<ModuleInfo>(info)));
   }
 
-  void addBuiltinModule(const kj::Path& specifier, kj::ArrayPtr<const char> sourceCode) {
+  void addBuiltinModule(kj::StringPtr specifier,
+                        kj::ArrayPtr<const char> sourceCode,
+                        Type type = Type::BUILTIN) {
     // Register new module accessible by a given importPath. The module is instantiated
     // after first resolve attempt within application has failed, i.e. it is possible for
     // application to override the module.
     // sourceCode has to exist while this ModuleRegistry exists.
     // The expectation is for this method to be called during the assembly of worker global context
     // after registering all user modules.
+    KJ_ASSERT(type != Type::BUNDLE);
+    using Key = typename Entry::Key;
 
-    if (entries.find(specifier) != nullptr) {
+    // We need to make sure there is not an existing worker bundle module with the same
+    // name if type == Type::BUILTIN
```

Comment:
Is this for optimization, or correctness? From reading resolveDynamicImport(), I guess we need to be able to answer the question, 'the module named "node:path" is builtin?' based on just what's in `entries`?

Reply:
It's for correctness. The idea is to only register the built in if there is not already a user-bundle entry of the same name